src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNICommon.cs

```diff
@@ -44,6 +44,46 @@ internal class SNISMUXHeader
         public uint length;
         public uint sequenceNumber;
         public uint highwater;
+
+        public void Read(byte[] bytes)
+        {
+            SMID = bytes[0];
+            flags = bytes[1];
```

Comment:
would inverting line 51 and 50 here allow for the same benefit discussed line 61 ?
> access the highest element first to cause the largest range check in the jit

Reply:
Yes, I decided it was unlikely to make a significant difference by the time i'd got to this one though. If later benchmarking shows this as a hot path i'll revisit and do the same, in the meantime the JIT constantly improves and may no longer need the reverse order layout to produce optimal code. We'll see.